core/embed/extmod/modtrezorui/display-stm32_T.h

```diff
@@ -56,6 +56,8 @@ const volatile uint8_t DISPLAY_ST7789V_INVERT_COLORS = 0;
 // of ILI9341V datasheet
 #define DISPLAY_ID_ILI9341V 0x009341U
 
+void display_pixeldata(uint16_t c) { PIXELDATA(c); }
```

Comment:
wondering if we could `inline` this function (and then `#[inline(always)]` Rust's `pixeldata()`

Reply:
I see, inlining across languages is problematic. There is something called Linker plugin LTO for Rust, but it's unclear if it even works with gcc.

But the function call overhead might hurt here. Wondering how best to expose the underlying thing to Rust.

Could we redefine `ADDR` to be something like `const _IO uint8_t* PIXELDATA_ADDR = (_IO uint8_t*)((uint32_t)...)`, then use the `PIXELDATA_ADDR` pointer manually?

Reply:
```rust
#[cfg(feature = "model_tt")]
#[inline(always)]
pub fn pixeldata(c: u16) {
    unsafe {
        PIXELDATA_ADDR.write_volatile((c >> 8) as u8);
        PIXELDATA_ADDR.write_volatile((c & 0xff) as u8);
    }
}

#[cfg(not(feature = "model_tt"))]
#[inline(always)]
pub fn pixeldata(c: u16) {
    unsafe {
        display_pixeldata(c);
    }
}
```

Reply:
[aea7578](https://github.com/trezor/trezor-firmware/pull/2317/commits/aea7578520e9bf2dc6fc6d923785fa0e0f3264d2) does the above